src/Multivariate/ProductFun.jl

```diff
@@ -74,6 +84,15 @@ function ProductFun(cfs::AbstractMatrix{T},sp::AbstractProductSpace{Tuple{S,V},D
     end
 end
 
+## TODO: This Product Fun actually does not return a productfun, dirty but was easiest to implement. Probably an abstract type of ProductFuns
+# is needed in the future.
+function ProductFun(iter::TrivialTensorizer{d},cfs::Vector{T},blk::Block, sp::AbstractProductSpace{<:NTuple{d, <:UnivariateSpace}}) where {T<:Number,d}
```

Comment:
This should probably add a method to `MultivariateFun` instead of `ProductFun`

Reply:
I am not sure why, it just overwrites ProductFun for the TrivialTensorizer case.

Reply:
A constructor for a struct should return a value of the same type. While it's possible to violate this, it's usually not a good idea. Do you specifically require `ProductFun` to me modified this way, to hit this method in the dispatch sequence somewhere?

Reply:
I will check if I can implement this in a better way. 
The problem is also that the ProductFun is only for bivariate functions and the implementation does not scale well to higher dimensions. Maybe one should completely separate the 2D case from higher dimensional cases or also rewrite the 2D case.
```
struct ProductFun{S<:UnivariateSpace,V<:UnivariateSpace,SS<:AbstractProductSpace,T} <: BivariateFun{T}
    coefficients::Vector{VFun{S,T}}     # coefficients are in x
    space::SS
end
```

Generally, the current order in which coefficients are stored is not ideal for the evaluation. In my opinion, the best thing would be to change this order.